File: leetcode/cpp/824_Goat_Latin.py
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        vowels = set("aeiouAEIOU")

        return " ".join(
            (
                (word + "ma" if word[0] in vowels
                 else word[1:] + word[0] + "ma")
                 + "a" * i
            )
            for i, word, in enumerate(sentence.split(), start=1)
        )

File: leetcode/cpp/824_Goat_Latin.cpp
#include <string>
#include <sstream>
#include <cctype>
using namespace std;
class Solution {
public:
	string toGoatLatin(string sentence) {
		stringstream ss(sentence);
		string word;

		string ret = "";
		int a = 1;
		while (ss >> word) {
            char c = word[0];
			switch(c) {
				case 'a':
				case 'e':
				case 'i':
				case 'o':
				case 'u':
				case 'A':
				case 'E':
				case 'I':
				case 'O':
				case 'U':
					word += "ma";	
                    break;
				default: 
					word.push_back(c);
                    word += "ma";
					word = word.substr(1);
			}

			for (int i = 0; i < a; ++i) 
				word.push_back('a');
            
            ++a;

			ret += (word + " ");
		}

        ret.pop_back();
		return ret;
	}
};




